### PR TITLE
net: lib: aws_iot: Initialize err

### PR DESCRIPTION
**Pull request by DurandA**
Fix uninitialized error code if CONFIG_AWS_FOTA is not set.

### CODE REVIEW

File: subsys/net/lib/aws_iot/src/aws_iot.c [hunk 1]
```diff
@@ -535,7 +535,7 @@ static int publish_get_payload(struct mqtt_client *const c, size_t length)
 static void mqtt_evt_handler(struct mqtt_client *const c,
 			     const struct mqtt_evt *mqtt_evt)
 {
-	int err;
+	int err = 0;
```
Comment (rlubos) [inline subsys/net/lib/aws_iot/src/aws_iot.c:538]: I see that there is one place where this variable could be used w/o initialization (at line 586), but it seems it's used wrongly in there (`res` should be used instead, or we could even give up on `res` variable and do `err = topic_subscribe();`, if I'm not mistaken). IMO it should be fixed in there instead.

**Review by simensrostad — APPROVED** [review]
LGMT

**Comment by DurandA** [comment]
> LGMT

What does it mean?

**Review by rlubos — APPROVED** [review]
LGTM (looks good to me)

**Comment by rlubos** [comment]
@DurandA Could you rebase please? We've just upmerged Zephyr and it's causing the CI to fail.

**Comment by DurandA** [comment]
Let me know if it still fails. Unfortunately the report is not available to external contributors.

**Comment by rlubos** [comment]
The CI failure is not related to this PR, merging.

**Pull request merged**